python/pyarrow/parquet.py

```diff
@@ -1006,30 +1092,47 @@ class ParquetDataset:
     Check that individual file schemas are all the same / compatible.
 filters : List[Tuple] or List[List[Tuple]] or None (default)
     List of filters to apply, like ``[[('x', '=', 0), ...], ...]``. This
-    implements partition-level (hive) filtering only, i.e., to prevent the
-    loading of some files of the dataset.
-
-    Predicates are expressed in disjunctive normal form (DNF). This means
-    that the innermost tuple describe a single column predicate. These
-    inner predicate make are all combined with a conjunction (AND) into a
-    larger predicate. The most outer list then combines all filters
-    with a disjunction (OR). By this, we should be able to express all
-    kinds of filters that are possible using boolean logic.
-
-    This function also supports passing in as List[Tuple]. These predicates
-    are evaluated as a conjunction. To express OR in predicates, one must
-    use the (preferred) List[List[Tuple]] notation.
+    implements partition-level (hive) filtering, i.e., to prevent the
+    loading of some files of the dataset, as well as file-level filtering
+    (if `use_legacy_dataset` is set to False).
```

Comment:
The description `like [[('x', '=', 0), ...], ...]` is now in `_DNF_filter_doc` which appears below, so we don't need to repeat it.
```suggestion
    Rows which do not match the filter will be removed from scanned data.
    Partition keys embedded in a hive-style directory structure will be
    exploited to avoid loading files at all if they contain no matching rows.
    If `use_legacy_dataset` is set to True, filters can only reference partition
    keys.
```